src/managers/sessionManager.ts

```diff
@@ -0,0 +1,164 @@
+interface Session {
+    sessionId: string;
+    creatorId: string;
+    maxPlayers: number;
+    minPlayers: number;
+    privateSlots: number;
+    scoreLimit: number;
+    timeLimit: number;
+    gameModeId: number;
+    eloRating: number;
+    regionId: number;
+    difficulty: number;
+    hasStarted: boolean;
+    enableVoice: boolean;
+    matchType: string;
+    maps: string[];
+    originalSessionId: string;
+    customSettings: string;
+    rewardSeed: number;
+    guildId: string;
+    buildId: number;
+    platform: number;
+    xplatform: boolean;
+    freePublic: number;
+    freePrivate: number;
+    fullReset: number;
+}
+
+const sessions: Session[] = [];
+
+function createNewSession(sessionData: Session, Creator: string): Session {
+    const sessionId = getNewSessionID();
+    const newSession: Session = {
+        sessionId,
+        creatorId: Creator,
+        maxPlayers: sessionData.maxPlayers || 4,
+        minPlayers: sessionData.minPlayers || 1,
+        privateSlots: sessionData.privateSlots || 0,
+        scoreLimit: sessionData.scoreLimit || 15,
+        timeLimit: sessionData.timeLimit || 900,
+        gameModeId: sessionData.gameModeId || 0,
+        eloRating: sessionData.eloRating || 3,
+        regionId: sessionData.regionId || 3,
+        difficulty: sessionData.difficulty || 0,
+        hasStarted: sessionData.hasStarted || false,
+        enableVoice: sessionData.enableVoice || true,
+        matchType: sessionData.matchType || "NORMAL",
+        maps: sessionData.maps || [],
+        originalSessionId: sessionData.originalSessionId || "",
+        customSettings: sessionData.customSettings || "",
+        rewardSeed: sessionData.rewardSeed || -1,
+        guildId: sessionData.guildId || "",
+        buildId: sessionData.buildId || 4920386201513015989,
+        platform: sessionData.platform || 0,
+        xplatform: sessionData.xplatform || true,
+        freePublic: sessionData.freePublic || 3,
+        freePrivate: sessionData.freePrivate || 0,
+        fullReset: 0
+    };
+    sessions.push(newSession);
+    return newSession;
+}
+
+function getAllSessions(): Session[] {
+    return sessions;
+}
+
+function getSessionByID(sessionId: string): Session | undefined {
+    return sessions.find(session => session.sessionId === sessionId);
+}
+
+interface FindSessionRequest {
```

Comment:
this is basically an 'any' type, could move it to another file an call it "LooseObject"